pkg/resources/table_grant.go

```diff
@@ -116,21 +114,34 @@ func CreateTableGrant(data *schema.ResourceData, meta interface{}) error {
 
 	// ID format is <db_name>|<schema_name>|<table_name>|<privilege>
 	// table_name is empty when on_future = true
-	if onFuture {
-		data.SetId(fmt.Sprintf("%v|%v||%v", dbName, schemaName, priv))
-	} else {
-		data.SetId(fmt.Sprintf("%v|%v|%v|%v", dbName, schemaName, tableName, priv))
+	grantID := &grantID{
+		ResourceName: dbName,
+		SchemaName:   schemaName,
+		Privilege:    priv,
+	}
+	if !onFuture {
+		grantID.ViewName = tableName
```

Comment:
this feels a bit wonky since we're assigning a `tableName` to a struct field named `ViewName`. Is there a more suitable name for this field other than `ViewName`?

Reply:
the view_grant.go file doesn't use tableName, so I'm not sure what I should call this attribute :/

Reply:
maybe `ViewOrTable` ?